### ISSUE TITLE
Possible incorrect behavior of `indent`

### ISSUE THREAD

**Issue by GirZ0n** [pinterest/ktlint#1553]
If I run ktlint on the following code:
```kotlin
fun main() {
    val n = 42
    if (n == 10)
    {
        println("abc")
    }
}
```

I get 4 violations: 1 from `curly-spacing` and 3 from `indent`.
```
Main.kt:4:1: Unexpected indentation (4) (should be 8) (indent)
Main.kt:4:5: Unexpected newline before "{" (curly-spacing)
Main.kt:5:1: Unexpected indentation (8) (should be 12) (indent)
Main.kt:6:1: Unexpected indentation (4) (should be 8) (indent)
```

## Expected Behavior
But I expect to get only the `curly-spacing` violation, since fixing it will fix the indentation problems as well.

## Your Environment
* Version of ktlint used: 0.46.1 via command line

**Comment by paul-dingemans** [comment]
This is inherently to the way ktlint works. When running without reformatting all rules are run simultaneously. So the indent rule just doesn't know about the violation found by the curly-spacing rule. From the perspective of the indent rule the code would be formatted well as follows:
```
fun main() {
    val n = 42
    if (n == 10)
        {
            println("abc")
        }
}
```
